bigbang/ingress/abstract.py

```diff
@@ -0,0 +1,773 @@
+                msg.set_param("Content-Type", value)
+            elif "mime-version" == key:
+                msg.set_param("MIME-Version", value)
+            elif "content-transfer-encoding" == key:
+                msg.set_param("Content-Transfer-Encoding", value)
+            else:
+                try:
+                    # TODO: find out why it sometimes raises
+                    # email/_header_value_parser.py
+                    # IndexError: list index out of range.
+                    # Also look into UTF-8 encoding.
+                    msg[key] = value
+                except Exception:
+                    pass
+        if (
+            (msg["Message-ID"] is None)
+            and (msg["Date"] is not None)
+            and (msg["From"] is not None)
+        ):
+            msg["Message-ID"] = archived_at.split("/")[-1]
+        # convert to `EmailMessage` to `mboxMessage`
+        mbox_msg = mboxMessage(msg)
+        mbox_msg.add_header("Archived-At", "<" + archived_at + ">")
+        return mbox_msg
+
+    def from_url(
+        self,
+        list_name: str,
+        url: str,
+        fields: str = "total",
+    ) -> mboxMessage:
+        """
+        Parameters
+        ----------
+        list_name : The name of the mailing list.
+        url : URL of this Email
+        fields : Indicates whether to return 'header', 'body' or 'total'/both or
+            the Email. The latter is the default.
+        """
+        soup = get_website_content(url, session=self.session)
+        if soup == "RequestException":
+            body = "RequestException"
+            header = self.empty_header
+        else:
+            if fields in ["header", "total"]:
+                header = self._get_header_from_html(soup)
+            else:
+                header = self.empty_header
+            if fields in ["body", "total"]:
+                body = self._get_body_from_html(list_name, url, soup)
+            else:
+                body = None
+        return self.create_email_message(url, body, **header)
+
+    @abstractmethod
+    def _get_header_from_html(self, soup: BeautifulSoup) -> Dict[str, str]:
+        pass
+
+    @abstractmethod
+    def _get_body_from_html(
+        self, list_name: str, url: str, soup: BeautifulSoup
+    ) -> Union[str, None]:
+        pass
+
+    @staticmethod
+    def to_dict(msg: mboxMessage) -> Dict[str, List[str]]:
+        """Convert mboxMessage to a Dictionary"""
+        return MessageIO.to_dict(msg)
+
+    @staticmethod
+    def to_pandas_dataframe(msg: mboxMessage) -> pd.DataFrame:
+        """Convert mboxMessage to a pandas.DataFrame"""
+        return MessageIO.to_pandas_dataframe(msg)
+
+    @staticmethod
+    def to_mbox(msg: mboxMessage, filepath: str):
+        """
+        Parameters
+        ----------
+        msg : The Email.
+        filepath : Path to file in which the Email will be stored.
+        """
+        return MessageIO.to_mbox(msg, filepath)
+
+
+class AbstractList(ABC):
```

Comment:
Maybe AbstractListScraper if it really has to do with the ingress of the list and not the representation of the list?

I'm unsure of how this class relates to ListIO